### DISCUSSION TITLE
All features zeroed but one when using CoxNet

### DISCUSSION THREAD

**Discussion by tomas-silveira** [sebp/scikit-survival#264] [Q&A] [answered]
Hello, so I'm trying to use the `CoxnetSurvivalAnalysis` with a grid search for several alphas. However, the best model that I'm getting is one with only one feature that's not zero. Here's the model that I'm running:

```
cv = KFold(n_splits=5, shuffle=True, random_state=SEED)
gcv = GridSearchCV(
    CoxnetSurvivalAnalysis(l1_ratio=0.9),
    param_grid={"alphas": [[v] for v in est_alphas]},
    cv=cv,
    error_score=0.5,
    verbose=2)\
    .fit(X_train, y_train)
```

For the set of alphas, I ran a smaller model similar to the one exemplified in the tutorial, with alphas ranging from 0.26 to 0.0026). My dataset is composed of numeric variables that were standardized prior to fitting the model. The resulting plot for the mean c-score for the set of alphas is also a bit odd, as you can see below:

<img width="565" alt="1" src="https://user-images.githubusercontent.com/41126502/165364226-5db742ba-ecb4-4f9d-8591-dd2c74935270.PNG">

The non-zero feature is one of the top10 features in a benchmark model, so it's quite strange that Im getting this results... Any ideas on how to improve? Thank you in advance.

**Comment by sebp** [comment, marked as answer]
Nothing obvious comes to my mind. What's the size of your data matrix 
(number of samples and features)?

**Reply by mldiscussion** [reply]
It appears we have encountered a similar issue and in our case, all features have coefficient of zero, with C-index of 0.49.  We suspect this is due to the dataset size and number of features: we have around 240 samples and each sample has around 35 features.  Could this be a reason?  The alpha is 0.035, so it is not particularly large. We used xgboost which seems working much better.

**Reply by tomas-silveira** [reply]
Thank you for your answers, ive since found what the error was, it had to do with something in my data.

**Comment by sebp** [comment]
It's very difficult for me to judge what is going on without an example 
dataset. Maybe you can create synthetic data that has properties similar 
to your data and causes the same issue?
